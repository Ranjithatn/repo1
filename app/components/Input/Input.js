import React from "react";
import Icon from "../Icon/Icon";

const Input = ({
  onChange,
  value,
  id,
  placeholder,
  name,
  type,
  maxLength,
  minLength,
  disabled = false,
  defaultValue,
  defaultChecked,
  className,
  style={},
  wrapperStyle={},
  onKeyPress,
  leftIcon,
  rightIcon,
  onClick,
  close,
  search
}) => {
  return (
    <div className="control" style={ wrapperStyle }>
    {leftIcon && <Icon icon={leftIcon} onClick={close}/>}
      <input
        id={id}
        className={className}
        name={name}
        type={type || "text"}
        value={value}
        placeholder={placeholder}
        onChange={onChange}
        maxLength={maxLength}
        minLength={minLength}
        disabled={disabled}
        defaultValue={defaultValue}
        defaultChecked={defaultChecked}
        style={style}
        onKeyPress={ onKeyPress ? onKeyPress : () => {} }
      />
      {rightIcon && <Icon icon={rightIcon} onClick={search}/>}
    </div>
  );
};

export default Input;
