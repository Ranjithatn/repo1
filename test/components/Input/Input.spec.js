import React from "react";
import toJson from "enzyme-to-json";
import Input from "../../../app/components/Input/Input";
import Adapter from "enzyme-adapter-react-16";
import { configure, shallow, mount } from "enzyme";


configure({ adapter: new Adapter() });


describe('<Input />', () => {

  it("should render without crashing", () => {
    const component = mount(
      <Input />
    );
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
