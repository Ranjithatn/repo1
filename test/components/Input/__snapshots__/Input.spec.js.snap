// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render without crashing 1`] = `
<Input>
  <div
    className="control"
    style={Object {}}
  >
    <input
      disabled={false}
      onKeyPress={[Function]}
      style={Object {}}
      type="text"
    />
  </div>
</Input>
`;
